packages/core/src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -1,89 +1,20 @@
-import { useEffect, useId, useRef } from 'react'
 import { useInView } from 'react-intersection-observer'
-
-import { ProductShelf as UIProductShelf } from '@faststore/ui'
-
-import type { ProductsQueryQueryVariables } from '@generated/graphql'
-import ProductShelfSkeleton from 'src/components/skeletons/ProductShelfSkeleton'
-import { useViewItemListEvent } from 'src/sdk/analytics/hooks/useViewItemListEvent'
-import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
-import { textToKebabCase } from 'src/utils/utilities'
-
-import Carousel from '../../ui/Carousel'
 import Section from '../Section'
-import { Components } from './Overrides'
-const { ProductCard } = Components
 
 import styles from './section.module.scss'
+import UIProductShelf, {
+  ProductShelfProps,
+} from 'src/components/ui/ProductShelf'
 
-interface ProductShelfProps extends Partial<ProductsQueryQueryVariables> {
-  title: string
-  withDivisor?: boolean
-}
-
-function ProductShelf({
-  title,
-  withDivisor = false,
-  ...variables
-}: ProductShelfProps) {
-  const titleId = textToKebabCase(title)
-  const id = useId()
-  const viewedOnce = useRef(false)
+function ProductShelf({ ...otherProps }: Omit<ProductShelfProps, 'inView'>) {
```

Comment:
Maybe we can call it `ProductShelfSection` and the one from 'src/components/ui/ProductShelf' as `ProductShelf` instead of `UIProductShelf`'?
*UI prefix means it comer from UI package.

Reply:
Right!